File: content/problems/2895-minimum-processing-time/solutions/opposite-sorting-in-groups-of-four/Solution.java
class Solution {
    public int minProcessingTime(List<Integer> processorTime,List<Integer> tasks){
        List<Integer> processors=new ArrayList<>(processorTime),sortedTasks=new ArrayList<>(tasks);
        Collections.sort(processors);
        sortedTasks.sort(Collections.reverseOrder());
        int answer=0;
        for(int i=0;i<processors.size();i++)answer=Math.max(answer,processors.get(i)+sortedTasks.get(4*i));
        return answer;
    }
}

File: content/problems/2895-minimum-processing-time/solutions/opposite-sorting-in-groups-of-four/Solution.kt
class Solution {
    fun minProcessingTime(processorTime:List<Int>,tasks:List<Int>):Int{
        val processors=processorTime.sorted();
        val sortedTasks=tasks.sortedDescending();
        var answer=0;
        for(i in processors.indices)answer=maxOf(answer,processors[i]+sortedTasks[4*i]);
        return answer
    }
}
